lib/machinery_helper.rb

```diff
@@ -34,33 +31,42 @@ def initialize(s)
     @system = s
     @arch = @system.arch
 
-    @local_helpers_path = "/usr/share/machinery/helpers"
+    @local_helpers_path = File.join(Machinery::ROOT, "machinery-helper")
+    @remote_helper_path = File.join(Machinery::REMOTE_HELPER_PATH, "machinery-helper")
   end
 
   def local_helper_path
-    File.join(@local_helpers_path, @arch, "machinery-helper")
+    File.join(@local_helpers_path, "machinery-helper")
   end
 
   # Returns true, if there is a helper binary matching the architecture of the
   # inspected system. Return false, if not.
   def can_help?
-    File.exist?(local_helper_path)
+    if File.exist?(local_helper_path) && LocalSystem.matching_architecture?(@arch)
+      true
+    else
+      false
+    end
   end
 
   def inject_helper
-    @system.inject_file(local_helper_path, Machinery::REMOTE_HELPERS_PATH)
+    @system.inject_file(local_helper_path, Machinery::REMOTE_HELPER_PATH)
```

Comment:
There is some confusion here about what a remote helper path is. Either it is the path to the binary (as in @remote_helper_path`) or it is the path to the directory where the helper is (as in `Machinery::REMOTE_HELPER_PATH`).

These should be two clearly distinguished names and they should use the same level of encapsulation (if one of them is encapsulated by a method, the other one should be as well).